R/occDetFunc.r

```diff
@@ -216,15 +241,46 @@ occDetFunc <- function (taxa_name, occDetdata, spp_vis, n_iterations = 5000, nyr
   # Convert the regional table to these numeric versions of site names
   if(!is.null(regional_codes)){
     regional_codes$numeric_site_name <- site_match$new_site_name[match(x = as.character(regional_codes[,1]),
-                                                             table = as.character(site_match$original_site))]
+                                                                       table = as.character(site_match$original_site))]
   }
   
-  # need to get a measure of whether the species was on that site in that year, unequivocally, in zst
-  zst <- acast(occDetdata, site ~ factor(year), value.var = 'focal', max, fill = 0) # initial values for the latent state = observed state
-  nyear <- max_year - min_year + 1
+
+  
+  # if the max_year is not null, edit the zst table to add the additional years required
+  if(!is.null(max_year)){
+    
+    # check that max_year is a numeric value
+    if(!is.numeric(max_year)) stop('max_year should be a numeric value')
+
+    # need to get a measure of whether the species was on that site in that year, unequivocally, in zst
+    zst <- acast(occDetdata, site ~ factor(year), value.var = 'focal', max, fill = 0) # initial values for the latent state = observed state
+    nyear <- max_year - min_year + 1
+    
+    # if nyear is greater than the number of years due to different specification of max_year, 
+    # add on additional columns to zst so that inital values can be create for these years.
+    
+    if(nyear > ncol(zst)){
+      # work out how many columns need to be added
+      to_add <- nyear - ncol(zst)
+      zst <- cbind(zst, matrix(0, ncol = to_add, nrow = nrow(zst)))
+      # add column names
+      colnames(zst) <- 1:nyear 
+    }
+    
+    # if a value has not been selected for max_year then continue as before
+  }else{
+    
+    # record the max year
+    max_year <- max(occDetdata$year)
```

Comment:
@CharlieOuthwaite I think I have it this time. Here max_year is defined as the max value of year AFTER year has been rescaled to start at 1 (L236). But min_year (used below) is defined on the original scale, which may start in 2010. I'm pretty sure that is it. the easiest fix would be to redefine min_year here to make sure they are both on the same scale

Reply:
As a result of this nyear gets a negative value, nyear is passed to initiate which get passed to rep where nyears is used as the `times` arguement... hence the error is about the times arguement

Reply:
Sorry, I had already figured this out haha!  I have just shifted the redefining of the years to numeric values to below where the max_year stuff is done.  I am just testing the changes now...